useeffect-and-calling-apis/calling-apis-from-components.md

```diff
@@ -0,0 +1,440 @@
+      <img src={imageUrl} alt="A random dog" />
+    </div>
+  </div>);
+}
+```
+<!-- prettier-ignore-end -->
+
+The `onClick` handler for this button is a one-line anonymous arrow function that logs "The button was clicked!" to the console.
+
+![](../assets/useeffect-and-calling-apis_calling-apis-from-components_button-console-log.png)
+
+### Event Handling With API Calls
+
+Now, instead of logging to the console, they want to call the API and update state appropriately.
+
+Belinda sets the `onClick` handler to be a new function, named `onButtonClick`. This function, `onButtonClick` will invoke another function, `getRandomImage`.
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+import axios from 'axios';
+
+function App() {
+  const [imageUrl, setImageUrl] = useState('https://images.dog.ceo/breeds/hound-afghan/n02088094_1003.jpg');
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const getRandomImage = () => {
+    axios.get('https://dog.ceo/api/breeds/image/random')
+      .then((response) => {
+        setImageUrl(response.data.message);
+        setErrorMessage('');
+      }).catch((error) => {
+        setErrorMessage(<section>{error.response.data.message}</section>);
+      });
+  };
+
+  const onButtonClick = () => {
+    getRandomImage();
+  };
+
+  return (<div>
+    <h1>My Dog Log</h1>
+    {errorMessage}
+    <div>
+      <button onClick={onButtonClick}>Get New Random Dog Image</button>
+      <img src={imageUrl} alt="A random dog" />
+    </div>
+  </div>);
+}
+```
+<!-- prettier-ignore-end -->
+
+Therefore, whenever a user clicks on the button, `onButtonClick` runs. This function in itself calls `getRandomImage`.
+
+![](../assets/useeffect-and-calling-apis_calling-apis-from-components_button-verify.png)
+
+![](../assets/useeffect-and-calling-apis_calling-apis-from-components_button-verify-dev-tools.png)
+
+### !callout-info
+
+## Defining Our Event Handler Inline
+
+The function `onButtonClick` is so small! Could we define our event handler inline?
+
+<!-- prettier-ignore-start -->
+```js
+import axios from 'axios';
+import { useState } from 'react';
+
+function App() {
+  const [imageUrl, setImageUrl] = useState('https://images.dog.ceo/breeds/hound-afghan/n02088094_1003.jpg');
+
+  const getRandomImage = () => {
+    axios.get('https://dog.ceo/api/breeds/image/random')
+      .then((response) => {
+        setImageUrl(response.data.message);
+        setErrorMessage('');
+      }).catch((error) => {
+        setErrorMessage(<section>{error.response.data.message}</section>);
+      });
+  };
+
+  return (<div>
+    <h1>My Dog Log</h1>
+    <div>
+      <button onClick={() => { getRandomImage(); }}>Get New Random Dog Image</button>
+      <img src={imageUrl} alt="A random dog" />
+    </div>
+  </div>);
+}
+```
+<!-- prettier-ignore-end -->
+
+Our `onClick` handler is now a one-line anonymous function! When this anonymous function runs, it will do one thing: invoke `getRandomImage`.
```

Comment:
spacing due to being in a details in a callout
```suggestion
<br/>

Our `onClick` handler is now a one-line anonymous function! When this anonymous function runs, it will do one thing: invoke `getRandomImage`.

</details>

```